### PR TITLE
Birthday picker fix

### PR DESCRIPTION
**Pull request by k3ntako**

### CODE REVIEW

File: client/pages/Me/BirthdayPicker.js [hunk 1]
```diff
@@ -67,6 +70,12 @@ export default class BirthdayPicker extends React.Component {
   sendChange(){
     if( this.state.year && this.state.month && this.state.date ){
       let birthday = this.props.value || new Date(0);
+
+      const lastDayOfMonth = MONTHS[this.state.month].last;
+      if( lastDayOfMonth < this.state.date ){
+        this.setState({ date: lastDayOfMonth }, this.sendChange)
+      }
```
Comment (k3ntako) [inline client/pages/Me/BirthdayPicker.js]: didn't account for leap year

File: client/components/ProfilePicture/index.css [hunk 2]
```diff
@@ -30,8 +30,8 @@
 .profile.tiny,
 .tiny > img,
 .tiny > span.initials {
-  width: 36px;
-  height: 36px;
+  width: 40px;
+  height: 40px;
```
Comment (k3ntako) [inline client/components/ProfilePicture/index.css]: Initials were too big on Roster. Fixed by making circle bigger.
Screen Shot 2019-06-05 at 4 40 56 PM

File: client/pages/Me/BirthdayPicker.js [hunk 3]
```diff
@@ -64,12 +67,28 @@ export default class BirthdayPicker extends React.Component {
...
+    const { date, month, year } = this.state;
+
+    const lastDayOfMonth = MONTHS[month].last;
+    const isLeapYear = year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0);
+    if( lastDayOfMonth < date && !isLeapYear ){ // not leap year
+      this.setState({ date: lastDayOfMonth }, this.sendChange);
+    }else if( month === 2 && 29 < date && isLeapYear ){ // leap year and February
+      this.setState({ date: 29 }, this.sendChange);
+    }
+  }
```
Comment (k3ntako) [inline client/pages/Me/BirthdayPicker.js]: make sure that the day exists in that month. For example, 30th doesn't exist in Feb.

File: client/pages/Review/index.js [hunk 4]
```diff
@@ -147,7 +147,7 @@ class ReviewPage extends React.Component {
 
         cells.push(<td key="name" className={styles.date}>
           <TooltipLink to={`/events/${event._id}/attendance`}
-              tooltip={event.name} 
+              tooltip={event.name + (event.is_cancelled ? ' [CANCELLED]' : '')}
```
Comment (k3ntako) [inline client/pages/Review/index.js]: Adds `[CANCELLED]` to tooltip, if event was cancelled.

File: client/pages/Team/EmailTour.js [hunk 5]
```diff
@@ -45,20 +45,6 @@ export default function EmailTour(props){
...
-    </strong>
-  </a>;
-
-  const atIndex = props.me.email.indexOf("@");
-  const userEmailDomain = atIndex > -1 ? props.me.email.slice(atIndex + 1) : "Gmail.com";
-  STEPS[2].content = [
-    <span key="part1" style={{'userSelect': 'none'}}>{`Using your email provider (e.g., ${userEmailDomain}), send an email to the team email address, `}</span>,
-    teamEmailHTML,
-    <span key="part3" style={{'userSelect': 'none'}}>, and your email will be sent to every coach and member on this team.</span>]
-
```
Comment (k3ntako) [inline client/pages/Team/EmailTour.js]: click to copy email from the tutorial wasn't working. It seemed pointless, so I removed it.

**Comment by k3ntako** [comment]
Roster - Initials were too big
UpdateMe - Birthday picker allowed user to enter dates that didn't exist
Review - Added `[CANCELLED]` to tooltip.
Email Tour - removed copying team email. It was broken and served almost no purpose.

File: client/pages/Me/BirthdayPicker.js [hunk 6]
```diff
@@ -64,12 +67,28 @@ export default class BirthdayPicker extends React.Component {
...
+    const { date, month, year } = this.state;
+    if( year && month && date ){
       let birthday = this.props.value || new Date(0);
-      birthday.setUTCFullYear( this.state.year );
-      birthday.setUTCMonth( this.state.month - 1 );
-      birthday.setUTCDate( this.state.date );
+
+      this.validate( birthday );
+
+      birthday.setUTCFullYear( year );
```
[the remark this answers is not in the document]
Comment (mienaikoe) [reply client/pages/Me/BirthdayPicker.js]: I would call do it async like:
onYearChange -> validate -> sendChange

All with setState callbacks that call the next step

**Pull request merged**